metron-platform/metron-indexing/src/main/java/org/apache/metron/indexing/dao/search/SearchDao.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.indexing.dao.search;
+
+import java.io.IOException;
+import java.util.List;
+import org.apache.metron.indexing.dao.update.Document;
+
+public interface SearchDao {
```

Comment:
Seems like if we're adding this (and I do like having this split out by functionality), it should also be applied to the Elasticsearch Daos (same for things like UpdateDao).

I'd also strongly be in favor of splitting out the ElasticsearchDao as part of this, to make sure that the overall structure of the two Dao sets are very comparable and easy to understand.  The integration tests should be able to insulate us from the obvious issues, and IMO it's worthwhile to do now.

Reply:
I was undecided as to whether we should do it in this PR or a future one and also wanted to make sure people liked it.  Will make this change.